### DISCUSSION TITLE
Question about plex server in tandem with DMB

### DISCUSSION THREAD

**Discussion by profucius** [I-am-PUID-0/DMB#69] [Q&A]
Hey there, I have a question about setting up my Plex server to use in tandem with DMB. I'll start with a bit of detail about my current setup, then ask the question after.

Previous to discovering DMB, I had been using Plex Server + Zurg on a Windows machine. It worked fine and continues to do so. Then I discovered Proxmox and used it to setup a homelab. I used Portainer as the manager to setup DMB, which got me running Zurg and Riven successfully.

However, I am still running my Plex Server + Zurg primarily on the Windows machine, primarily because I've been confused about which method would be best for running Plex for full transcode ability. I've since learned that I could either setup an LXC or a Docker container within Portainer to run Plex, and then pass GPU to it for transcoding.

The main issue I'm running into that makes me feel the need to move Plex from Windows is that Riven does not trigger Plex to rescan as I have it set up now, which means Riven keeps re-adding the same shows and movies that have already been downloaded, because Plex hasn't scanned yet. I've been doing it manually, and on a schedule nightly, but I need it to be automatic for this issue to be resolved.

But since DMB is running in a Portainer stack, I wanted to ask here what my best plan of action would be. I see in the DMB readme it says "The Plex server must be started after the rclone mount is available." Does this mean that I would need to install the Plex server in a stack beside DMB? Or could I still have Plex installed in an LXC (beside Portainer, in its own LXC)?

**Comment by I-am-PUID-0** [comment]
Apologies, I just saw this. 

If Plex is running on Windows or natively in Linux, you may not need to restart Plex each time DMB restarts — it seems to be limited to the containerized version of Plex. 

As for Riven and Plex interactions, I don't believe Riven relies upon Plex to determine the available content, as plex_debrid has done in the past. I believe Riven uses the database to track the state of each item. 

If Plex is containerized, then it's best to put it into the same Portainer Stack with DMB and utilize the depends_on function for the Plex server as shown here: https://github.com/I-am-PUID-0/DMB#-example-plex-docker-compose

If running Plex in a separate LXC, then I believe you would need to find a way to coordinate the restart/startup of Plex each time DMB is started/restarted.
